packages/web/components/Quest/QuestCompletions.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  Avatar,
+  Box,
+  HStack,
+  MetaButton,
+  Text,
+  useToast,
+  VStack,
+} from '@metafam/ds';
+import { MetaLink } from 'components/Link';
+import {
+  Player,
+  Quest,
+  QuestCompletionStatus_ActionEnum,
+  QuestCompletionStatus_Enum,
+  QuestWithCompletionFragmentFragment,
+  useUpdateQuestCompletionMutation,
+} from 'graphql/autogen/types';
+import moment from 'moment';
+import React, { useCallback, useState } from 'react';
+import { FaExternalLinkAlt } from 'react-icons/fa';
+
+import { useUser } from '../../lib/hooks';
+import { getPlayerName } from '../../utils/playerHelpers';
+import { ConfirmModal } from '../ConfirmModal';
+import { CompletionStatusTag } from './QuestTags';
+
+interface AlertSubmission {
+  status: QuestCompletionStatus_ActionEnum;
+  quest_completion_id: string;
+}
+
+type Props = {
+  quest: QuestWithCompletionFragmentFragment;
+};
+
+export const QuestCompletions: React.FC<Props> = ({ quest }) => {
+  const { user } = useUser();
+  const toast = useToast();
+  const [
+    alertSubmission,
+    setAlertSubmission,
+  ] = useState<AlertSubmission | null>(null);
+  const [
+    updateQuestCompletionStatus,
+    updateQuestCompletion,
+  ] = useUpdateQuestCompletionMutation();
+  const isMyQuest = user?.id === (quest as Quest).player.id;
+
+  const onConfirmAlert = useCallback(() => {
+    if (!alertSubmission) return;
+
+    updateQuestCompletion({
+      quest_completion_id: alertSubmission.quest_completion_id,
+      status: alertSubmission.status,
+    }).then((response) => {
+      if (response.data?.updateQuestCompletion?.success) {
+        toast({
+          title: 'Quest completion updated',
+          description: `The completion is now ${alertSubmission.status}`,
+          status: 'success',
+          isClosable: true,
+          duration: 4000,
+        });
+        setAlertSubmission(null);
+      } else {
+        toast({
+          title: 'Error while updating completion',
+          description:
+            response.error?.message ||
+            response.data?.updateQuestCompletion?.error ||
+            'unknown error',
+          status: 'error',
+          isClosable: true,
+          duration: 10000,
+        });
+      }
+    });
+  }, [alertSubmission, updateQuestCompletion, toast]);
+
+  return (
+    <Box>
+      <VStack spacing={4}>
+        {quest.quest_completions.length === 0 &&
+        <Text>
+          There are no proposal for this quest yet.
+        </Text>
+        }
+        {quest.quest_completions.map((questCompletion) => (
+          <Box key={questCompletion.id} w="100%">
+            <HStack px={4} py={4}>
+              <Avatar name={getPlayerName(questCompletion.player as Player)} />
```

Comment:
this avatar doesn't seem to render

Reply:
The src is not set here, done that on purpose.

We actually have a bigger problem:
The data here is coming from the query GetQuestWithCompletions which gets a quest and all its completions within a single graphql request. For each completions, we get the associated player.
But the avatar is not stored on our database, so we'd have to query 3Box for every player to get their profile picture in the quest request. I didn't want to do that.  
If you think having to call 3box for every quest completion by the remote schema directly on this request, I can add this.